### PR TITLE
Medical Organisation Pack - Roller Bed Racks

### PR DESCRIPTION
**Pull request by 1Sparky1**
Adds a storage rack structure for roller beds. Each rack can hold up to four beds.

racks

WIP until mapping freeze is over. Reliant on #17074 anyway for colour scheme.

Edit: Forgot image

### CODE REVIEW

**Comment by 1Sparky1** [comment]
!wip

**Comment by 1Sparky1** [comment]
!review

**Review by Arrow768 — APPROVED** [review]

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 1]
```diff
@@ -633,3 +635,67 @@
...
+	icon_state = "holder"
+	var/list/held = list()
+	var/initial_beds = 4
+
+/obj/structure/roller_rack/New()
+	..()
+	var/i
+	for(i=1, i<=initial_beds, i++)
+		var/obj/item/roller/RB = new /obj/item/roller(src)
+		LAZYADD(held, RB)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:652]: Held is not a lazylist

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 2]
```diff
@@ -633,3 +635,67 @@
...
+
+/obj/structure/roller_rack
+	name = "roller bed rack"
+	desc = "A rack for holding collapsed roller beds."
+	icon = 'icons/obj/rollerbed.dmi'
+	icon_state = "holder"
+	var/list/held = list()
+	var/initial_beds = 4
+
+/obj/structure/roller_rack/New()
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:647]: Initialize()

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 3]
```diff
@@ -633,3 +635,67 @@
 	R.add_fingerprint(user)
 	qdel(held)
 	held = null
+
+/obj/structure/roller_rack
+	name = "roller bed rack"
+	desc = "A rack for holding collapsed roller beds."
+	icon = 'icons/obj/rollerbed.dmi'
+	icon_state = "holder"
+	var/list/held = list()
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:644]: This list needs to be cleared and the contents qdeled in destroy()

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 4]
```diff
@@ -633,3 +635,67 @@
...
+
+	if(istype(W, /obj/item/roller))
+		var/obj/item/roller/RB = W
+
+		if(LAZYLEN(held) >= 4)
+			to_chat(user, SPAN_NOTICE("The rack has no space for \the [RB]"))
+			return
+
+		user.drop_from_inventory(RB, src)
+		LAZYADD(held, RB)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:699]: Held is not a lazylist

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 5]
```diff
@@ -633,3 +635,67 @@
...
+	. = ..()
+
+/obj/structure/roller_rack/attack_hand(mob/user)
+	if(!LAZYLEN(held))
+		to_chat(user, SPAN_NOTICE("The rack is empty."))
+		return
+
+	var/obj/item/roller/RB = held[LAZYLEN(held)]
+	user.put_in_hands(RB)
+	LAZYREMOVE(held, RB)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:682]: Held is not a lazylist

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 6]
```diff
@@ -633,3 +635,67 @@
...
+/obj/structure/roller_rack/examine(mob/user)
+	desc = "[initial(desc)] \nIt is holding [LAZYLEN(held)] beds."
+	. = ..()
+
+/obj/structure/roller_rack/attack_hand(mob/user)
+	if(!LAZYLEN(held))
+		to_chat(user, SPAN_NOTICE("The rack is empty."))
+		return
+
+	var/obj/item/roller/RB = held[LAZYLEN(held)]
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:680]: Held is not a lazylist

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 7]
```diff
@@ -633,3 +635,67 @@
...
+		I.pixel_x = (5 * beds)
+		beds++
+		add_overlay(I)
+
+/obj/structure/roller_rack/examine(mob/user)
+	desc = "[initial(desc)] \nIt is holding [LAZYLEN(held)] beds."
+	. = ..()
+
+/obj/structure/roller_rack/attack_hand(mob/user)
+	if(!LAZYLEN(held))
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:676]: Held is not a lazylist

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 8]
```diff
@@ -633,3 +635,67 @@
...
+	cut_overlays()
+	var/beds = 0
+	for(var/obj/item/roller/RB in held)
+		var/image/I = overlay_image(icon, "[icon_state]_bed_[RB.base_icon]")
+		I.pixel_x = (5 * beds)
+		beds++
+		add_overlay(I)
+
+/obj/structure/roller_rack/examine(mob/user)
+	desc = "[initial(desc)] \nIt is holding [LAZYLEN(held)] beds."
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm:672]: Held is not a lazylist

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Comment by NonQueueingMatt** [comment]
Reopen when ready.

**Pull request closed without merging**